### PR TITLE
Fix `Block`  dump in author lint

### PR DESCRIPTION
**Pull request by oli-obk**
changelog: The `#[clippy::author]` attribute now emits correct pattern code for block expressions

### CODE REVIEW

File: tests/ui/author/blocks.stdout [hunk 1]
```diff
@@ -0,0 +1,13 @@
+if_chain! {
+    if let ExprKind::Block(ref block) = expr.node;
+    if let Some(trailing_expr) = &block.expr;
+    if block.stmts.len() == 0;
+    then {
+        // report your lint here
+    }
+}
+if_chain! {
+    then {
```
Comment (flip1995) [inline tests/ui/author/blocks.stdout]: Is this empty `if_chain` intended?

File: tests/ui/author/blocks.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#![feature(stmt_expr_attributes)]
```
Comment (flip1995) [inline tests/ui/author/blocks.rs]:
```suggestion
#![rustfmt::skip]
#![feature(stmt_expr_attributes)]
```

File: tests/ui/author/blocks.stdout (hunk 1 above)
Comment (oli-obk) [reply tests/ui/author/blocks.stdout]: yea, it exists for the `foo` function which we don't handle yet. So it's mainly a canary.

File: tests/ui/author/blocks.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#![rustfmt::skip]
```
Comment (flip1995) [inline tests/ui/author/blocks.rs]: `error[E0658]: non-builtin inner attributes are unstable`

https://travis-ci.com/rust-lang/rust-clippy/jobs/201506944#L1272

Sorry always forget about this... The `rustfmt` needs to be added to the main function.

**Comment by flip1995** [comment]
Argggh...

@bors retry (apple timeout)

**Pull request merged**